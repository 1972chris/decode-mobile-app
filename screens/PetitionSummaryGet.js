import React from 'react';
import { Constants, WebBrowser } from 'expo';
import { StyleSheet, Text, View, Button, Linking } from 'react-native';
import PropTypes from 'prop-types';
import Router from '../Router';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: '#FFF',
    flex: 1,
    paddingTop: 30,
  },
  logo: {
    height: 60,
    marginBottom: 30,
    width: 80,
  },
  textSubHeading: {
    alignSelf: 'flex-start',
    fontSize: 18,
    marginBottom: 10,
    marginLeft: 10,
  },
  textContainer: {
    marginBottom: 30,
    padding: 10,
  },
  flatList: {
    marginTop: 15,
    marginBottom: 30,
  },
  buttonContainer: {
    width: 300,
  },
  textParagraph: {
    marginBottom: 15,
  },
  petitionSummary: {
    backgroundColor: '#EEE',
    width: 300,
  },
  petitionHeading: {
    fontSize: 14,
    color: '#9B9B9B',
  },
});

export default class PetitionSummaryGet extends React.Component {
  static route = {
    navigationBar: {
      tintColor: 'rgb(0,163,158)',
      title: 'Sign Petition',
    },
  }

  constructor(props) {
    super(props);
    this.goToPetitionSummarySign = this.goToPetitionSummarySign.bind(this);
  }

  goToPetitionSummarySign() {
    this.props.navigator.push(Router.getRoute('petitionSummarySign'));
  }

  removeLinkingListener = () => {
    Linking.removeEventListener('url', this.handleRedirect);
  };

  addLinkingListener = () => {
    Linking.addEventListener('url', this.handleRedirect);
  };

  handleRedirect = () => {
    WebBrowser.dismissBrowser();
    this.goToPetitionSummarySign();
  };

  openWebBrowserAsync = async () => {
    this.addLinkingListener();
    const result = await WebBrowser.openBrowserAsync(
      `http://localhost:3000/?linkingUri=${Constants.linkingUri}`,
    );
    this.removeLinkingListener();
    this.setState({ result });
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.textSubHeading}>Petition Summary</Text>
        <View style={styles.petitionSummary}>
          <Text style={styles.petitionHeading}>Title</Text>
          <Text>Create communal space in Atlantis</Text>
          <Text style={styles.petitionHeading}>Created by</Text>
          <Text>John Bloggs, Atlantis Community College</Text>
          <Text style={styles.petitionHeading}>Closing date</Text>
          <Text>31 August 2020</Text>
        </View>
        <View style={styles.textContainer}>
          <Text style={styles.textParagraph}>To sign this petition you must be a resident
            of Atlantis. To demonstrate that you are a resident of Atlantis,
            you are required to share the
            following attribute with Secure Petitions when you sign the petition.</Text>

          <Text>You do not currently have this attribute
            in your wallet. Select
            Get Attribute below to be guided through the process of
            verifying your Atlantis Residency with
            Atlantis Council. You will only have to do this once,
            after which your Atlantis Residency attribute will
            be stored in your wallet ready for next time you need it.</Text>
        </View>
        <View style={styles.buttonContainer}>
          <Button
            onPress={this.openWebBrowserAsync}
            title="Get Attribute"
            color="rgb(0,163,158)"
          />
        </View>
      </View>
    );
  }
}

PetitionSummaryGet.propTypes = {
  navigator: PropTypes.shape({ push: PropTypes.func.isRequired }),
};

PetitionSummaryGet.defaultProps = {
  navigator: '',
};
