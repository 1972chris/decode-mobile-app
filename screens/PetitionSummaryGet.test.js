import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import PetitionSummaryGet from './PetitionSummaryGet';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore([thunk]);

const initialState = {
  petitionLink: {
    petitionLink: 'http://something.com',
  },
  petition: {
    petition: {
      title: 'hello',
      description: 'world',
      closingDate: 'today',
    },
  },
};

it('renders PetitionSummaryGet component', () => {
  const wrapper = shallow(
    <PetitionSummaryGet />,
    { context: { store: mockStore(initialState) } },
  );
  expect(wrapper.dive()).toMatchSnapshot();
});
