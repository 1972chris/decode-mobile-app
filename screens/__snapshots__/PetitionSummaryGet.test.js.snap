// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PetitionSummaryGet component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(246, 246, 246)",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#FFF",
            "margin": 16,
            "padding": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": 20,
              "fontWeight": "500",
              "marginBottom": 20,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": 14,
              "lineHeight": 20,
              "marginBottom": 15,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontSize": 12,
            }
          }
        >
          Closing date: 
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "500",
            "marginBottom": 10,
            "marginLeft": 16,
          }
        }
      >
        Your Information
      </Text>
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#FFF",
            "padding": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          Verified Atlantis Resident*
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "marginVertical": 4,
            }
          }
        >
          To get this attribute you will be directed to the Atlantis Council website
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "backgroundColor": "rgb(0,163,158)",
              "borderRadius": 2,
              "elevation": 2,
              "height": 36,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
              "shadowColor": "rgba(0, 0, 0, 0.54)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 2,
              "width": 100,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "rgb(255, 255, 255)",
                "fontSize": 14,
                "fontWeight": "500",
              }
            }
          >
            REQUEST
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 12,
            "marginLeft": 16,
            "marginTop": 10,
          }
        }
      >
        *Required fields
      </Text>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderColor": "rgba(0, 0, 0, 0.15)",
        "borderWidth": 1,
        "height": 64,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "rgba(0,163,158,0.4)",
          "borderRadius": 2,
          "elevation": 2,
          "height": 36,
          "marginBottom": 18,
          "marginHorizontal": 16,
          "marginTop": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
          "shadowColor": "rgba(0, 0, 0, 0.54)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "rgba(255, 255, 255, 0.6)",
            "fontSize": 14,
            "fontWeight": "500",
          }
        }
      >
        SIGN PETITION
      </Text>
    </View>
  </View>
</View>
`;
